ush/setup.py

```diff
@@ -134,159 +118,74 @@ def load_config_for_setup(ushdir, default_config, user_config):
-    expt_basedir = cfg_d["workflow"]["EXPT_BASEDIR"]
+    expt_basedir = default_config["workflow"]["EXPT_BASEDIR"]
     if (not expt_basedir) or (expt_basedir[0] != "/"):
-        expt_basedir = os.path.join(homedir, "..", "expt_dirs", expt_basedir)
-    try:
-        expt_basedir = os.path.realpath(expt_basedir)
-    except:
-        pass
-    cfg_d["workflow"]["EXPT_BASEDIR"] = os.path.abspath(expt_basedir)
-
-    cfg_d.dereference()
-
-    # Do any conversions of data types
-    for sect, settings in cfg_d.items():
-        for k, v in settings.items():
-            if not (v is None or v == "") and isinstance(v, str):
-                cfg_d[sect][k] = str_to_list(v)
-
-    # Mandatory variables *must* be set in the user's config or the machine file; the default value is invalid
-    mandatory = [
-        "NCORES_PER_NODE",
-        "FIXgsm",
-        "FIXaer",
-        "FIXlut",
-        "FIXorg",
-        "FIXsfc",
-    ]
-    flat_cfg = flatten_dict(cfg_d)
-    for val in mandatory:
-        if not flat_cfg.get(val):
-            raise Exception(
-                dedent(
-                    f"""
-                    Mandatory variable "{val}" not found in:
-                    user config file {user_config}
-                                  OR
-                    machine file {machine_file} 
-                    """
-                )
-            )
+        expt_basedir = homedir.parent / "expt_dirs" / expt_basedir
+    default_config["workflow"]["EXPT_BASEDIR"] = str(Path(expt_basedir).resolve())
 
-    # Check that input dates are in a date format
-    dates = ["DATE_FIRST_CYCL", "DATE_LAST_CYCL"]
-    for val in dates:
-        if not isinstance(cfg_d["workflow"][val], datetime.date):
-            raise Exception(
-                dedent(
-                    f"""
-                            Date variable {val}={cfg_d['workflow'][val]} is not in a valid date format.
+    # Expand out the workflow tasks now that all settings have been applied
+    taskgroups = default_config["workflow"]["taskgroups"]
+    default_config["rocoto"]["tasks"] = {}
+    for taskgroup in taskgroups:
+        tasks = get_yaml_config(homedir / taskgroup)
+        keep = {k: v for k, v in tasks.items() if not re.search(r"^default_*", k)}
+        default_config["rocoto"]["tasks"].update(keep)
 
-                            For examples of valid formats, see the Users' Guide.
-                            """
-                )
-            )
+    # Update one more time in case there are user or machine settings to override the tasks
+    for cfg in (machine_config, user_config):
+        default_config.update_from(cfg)
+
+    # Dereference all Jinja expressions
+    default_config.dereference()
+
+    ## Do any conversions of data types
+    #for sect, settings in default_config.items():
+    #    for k, v in settings.items():
+    #        if not (v is None or v == "") and isinstance(v, str):
+    #            default_config[sect][k] = str_to_list(v)
```

Comment:
I should probably get rid of this. I left it in case the job was too big when SRW wasn't doing a whole bunch of type converting. It seems fine so far.